tests/api.c

```diff
@@ -34704,7 +34704,163 @@ static void test_wolfSSL_OCSP_get0_info()
     X509_free(issuer);
 
     printf(resultFmt, "passed");
-#endif /* OPENSSL_EXTRA & HAVE_OCSP */
+#endif
+}
+
+static void test_wolfSSL_i2d_OCSP_CERTID()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP)
+    printf(testingFmt, "wolfSSL_i2d_OCSP_CERTID()");
+
+    WOLFSSL_OCSP_CERTID certId;
+    byte* targetBuffer;
+    byte* beginTargetBuffer;
+    // OCSP CertID bytes taken from PCAP
+    byte rawCertId[] = {
+        0x30, 0x49, 0x30, 0x09, 0x06, 0x05, 0x2b, 0x0e, 0x03, 0x02, 0x1a, 0x05,
+        0x00, 0x04, 0x14, 0x80, 0x51, 0x06, 0x01, 0x32, 0xad, 0x9a, 0xc2, 0x7d,
+        0x51, 0x87, 0xa0, 0xe8, 0x87, 0xfb, 0x01, 0x62, 0x01, 0x55, 0xee, 0x04,
+        0x14, 0x03, 0xde, 0x50, 0x35, 0x56, 0xd1, 0x4c, 0xbb, 0x66, 0xf0, 0xa3,
+        0xe2, 0x1b, 0x1b, 0xc3, 0x97, 0xb2, 0x3d, 0xd1, 0x55, 0x02, 0x10, 0x01,
+        0xfd, 0xa3, 0xeb, 0x6e, 0xca, 0x75, 0xc8, 0x88, 0x43, 0x8b, 0x72, 0x4b,
+        0xcf, 0xbc, 0x91
+    };
+    int ret;
+    
+    XMEMSET(&certId, 0, sizeof(WOLFSSL_OCSP_CERTID));
+    certId.rawCertId = rawCertId;
+    certId.rawCertIdSize = sizeof(rawCertId);
+    targetBuffer = XMALLOC(sizeof(rawCertId), NULL, DYNAMIC_TYPE_TMP_BUFFER);
+
+    beginTargetBuffer = targetBuffer;
+    ret = wolfSSL_i2d_OCSP_CERTID(&certId, &targetBuffer);
+    // If target buffer is not null, function increments targetBuffer to point
+    // just past the end of the encoded data.
+    AssertPtrEq(targetBuffer, (beginTargetBuffer + sizeof(rawCertId)));
+    // Function returns the size of the encoded data.
+    AssertIntEQ(ret, sizeof(rawCertId));
+    for (int i = 0; i < ret; ++i)
+    {
+        AssertIntEQ(beginTargetBuffer[i], rawCertId[i]);
+    }
+
+    XFREE(beginTargetBuffer, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    targetBuffer = NULL;
+    ret = wolfSSL_i2d_OCSP_CERTID(&certId, &targetBuffer);
+    // If target buffer is null, function allocates memory for a buffer and
+    // copies the encoded data into it. targetBuffer then points to the start of
+    // this newly allocate buffer.
+    AssertIntEQ(ret, sizeof(rawCertId));
+    for (int i = 0; i < ret; ++i)
+    {
+        AssertIntEQ(targetBuffer[i], rawCertId[i]);
+    }
+
+    XFREE(targetBuffer, NULL, DYNAMIC_TYPE_OPENSSL);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_SINGLERESP_get0_id()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP)
+    printf(testingFmt, "wolfSSL_OCSP_SINGLERESP_get0_id()");
+
+    WOLFSSL_OCSP_SINGLERESP single;
+    const WOLFSSL_OCSP_CERTID* certId = wolfSSL_OCSP_SINGLERESP_get0_id(&single);
+    AssertPtrEq(&single, certId);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_single_get0_status()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP) 
+    printf(testingFmt, "wolfSSL_OCSP_single_get0_status()");
```

Comment:
This printf needs to be moved below variable declarations, otherwise some compilers will complain.